### PR TITLE
Dept. Specific ID cards & budget cards.

### PR DESCRIPTION
**Pull request by Agameofscones**
## About The Pull Request

Cherry picks and cleans up the commit history of my previous accidental PR https://github.com/tgstation/tgstation/pull/44947 towards TG master, rather than my intended downstream.

Also adds more job specific ID cards along with adding new ID sprites to role and event uniform datums where applicable.

image

Also I'm not a coder, so this is probably sooper spaghetti.

## Why It's Good For The Game

A lot of other servers have job specific ID cards. Helps in quick identification of access tied to cards or job identification of unknown individuals.

A nice bit of QOL I'd think.

## Changelog
:cl: Triiodine
imageadd: "Added department and job specific ID cards and budget cards.
spellcheck: "Updated the descriptions of some ID cards."
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
these are really nice changes, will miss the old look of the emag though. silver should maybe stand out more as an elevated position id

**Comment by plapatin** [comment]
3/4 sprite bad

**Comment by tralezab** [comment]
none of those are 3/4

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -326,8 +326,44 @@ update_label("John Doe", "Clowny")
 				return
 			assignment = u
 			update_label()
+
+			if("i") //icon_states.
+				var/list/i = list(
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/cards_ids.dm]: static this

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Needs a non-antagonist way to change the icon of this.

It would be nice if it was limited to the respective heads rather than the hop as a way to show "true access" rather than HOP given access.

**Comment by zxaber** [comment]
No robo card/10

But for real, why give the detective and warden atmos their own design but not robo, chem, viro, or especially genetics?

**Comment by PKPenguin321** [comment]
didn't we already have horizontal sprites, then we upgraded to the diagonal ones we have now? this feels like a step backwards

**Comment by GravitronBlue** [comment]
Not sure I'm the only one who has a problem with this or what the PR author from #21191 thinks, but...
> The hack-o'-lantern is a festive emag

Before:
image
After:
Hack-o-nope
The hack-o-lantern is now just an emag with an engraving/picture of a pumpkin on it.

**Comment by tralezab** [comment]
yeah, definitely miss the old hack-o-lantern. pls fix!

**Comment by PKPenguin321** [comment]
i think it would look nicer and be WAY cooler if we were to greyscale our current ID card sprites, then associate different access types with color values that are put on the ID dynamically. IE sec is red, medical is blue, so if you give a sec ID medbay access it becomes more purple (as red + blue = purple) the more medbay access it has

then you could also just add a button to the HoP console that can manually change the ID color if you want to disguise it

**Comment by Twaticus** [comment]
gold is not nearly gold enough and it looks really sad compared to our current version. the normal ID's look more silver than the actual silver ID imo

**Comment by Dorsisdwarf** [comment]
the overshading on these sprites makes them look like they're really really thick instead of ID cards

**Comment by cacogen** [comment]
I don't mind the artstyle but the colour schemes make no sense. The stripe should indicate department (so be green on mime and clown) and the circle should indicate the specific role in that department if need be (which is what you've done with some but not all jobs with their own colour schemes). I don't get the bit on the end and it should probably be consistent with the base card sprite.

**Comment by Dorsisdwarf** [comment]
image

**Comment by Alek2ander** [comment]
Proposal - use HUD icons:
id_preview

**Comment by Fikou** [comment]
SOULLESS

**Comment by Fikou** [comment]
> Proposal - use HUD icons:
> id_preview
> 
> [card.zip](https://github.com/tgstation/tgstation/files/3369415/card.zip)

pr this please

**Comment by Agameofscones** [comment]
> i think it would look nicer and be WAY cooler if we were to greyscale our current ID card sprites, then associate different access types with color values that are put on the ID dynamically. IE sec is red, medical is blue, so if you give a sec ID medbay access it becomes more purple (as red + blue = purple) the more medbay access it has
> 
> then you could also just add a button to the HoP console that can manually change the ID color if you want to disguise it

This is much beyond my coding capabilities, but I will be making the other changes requested along with  @Alek2ander 's IDs

**Comment by Alek2ander** [comment]
@Agameofscones @Fikou I'll make a PR with my sprite variations.

**Pull request closed without merging**